app/controllers/RaceGroupController.java

```diff
@@ -0,0 +1,112 @@
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import static play.libs.Json.toJson;
+
+public class RaceGroupController extends Controller {
+    private final RaceGroupRepository raceGroupRepository;
+
+    @Inject
+    public RaceGroupController(RaceGroupRepository raceGroupRepository) {
+        this.raceGroupRepository = raceGroupRepository;
+    }
+
+    public CompletionStage<Result> getAllRaceGroups() {
+        return raceGroupRepository.getAllRaceGroups().thenApplyAsync(raceGroups -> ok(toJson(raceGroups.collect(Collectors.toList())))).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()) {
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No racegroups are set in db");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRaceGroup(long id) {
+        return raceGroupRepository.getRaceGroupById(id).thenApplyAsync(raceGroup -> ok(toJson(raceGroup))).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NoResultException":
+                    res = badRequest("No racegroup with id: " + id + " is available in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> deleteAllRaceGroups() {
+        return raceGroupRepository.deleteAllRaceGroups().thenApply(raceGroups -> ok(toJson(raceGroups.collect(Collectors.toList())) + "have been deleted")).exceptionally(ex -> internalServerError(ex.getMessage()));
+    }
+
+    public CompletionStage<Result> deleteRaceGroup(long id) {
+        return raceGroupRepository.deleteRaceGroupById(id).thenApplyAsync(racegroup -> ok(toJson(racegroup) + "has been deleted")).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()) {
+                case "NoResultException":
+                    res = badRequest("Racegroup with id" + id + " not found in DB");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> addRaceGroup() {
+        JsonNode json = request().body().asJson();
+        return parseRaceGroup(json).thenApply(raceGroupRepository::addRaceGroup).thenApply(raceGroup -> ok(toJson(raceGroup) + " has been added")).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)